### DISCUSSION TITLE
trying to set up CI/CD with github getting message "No serverless.yml found in selected repository"

### DISCUSSION THREAD

**Discussion by geopopos** [serverless/serverless#11253] [Q&A]
Hey all,

I'm trying to set up ci/cd in one of my serverless projects.

When I go to set it up I get the message:
"No serverless.yml found in selected repository"

However, my serverless.yml file is there clear as day. You can see for yourself here: https://github.com/geopopos/aws_lasso_to_notion/blob/main/serverless.yml

Any idea why I'm getting this error?

Thanks!
George

**Comment by pgrzesik** [comment]
Hey @geopopos - are you talking about Dashboard's CI/CD offering or you're setting up your own CI/CD e.g. with Github Actions?

**Reply by geopopos** [reply]
@pgrzesik thanks for asking! I am using the dashboard's CI/CD offering

**Reply by pgrzesik** [reply]
Thanks for clarification @geopopos, then @garethmcc might be able to help a bit more than me 🙇

**Reply by geopopos** [reply]
Appreciate it! @pgrzesik 
@garethmcc let me know what details I can provide that would be helpful to you :)

**Reply by pgrzesik** [reply]
The answer from @masanaikeshima is correct, there's an issue with Github's indexing that is causing it

**Reply by geopopos** [reply]
@pgrzesik my repository was private. I tried changing it to public to see if that would work

**Reply by geopopos** [reply]
@pgrzesik @masanaikeshima The repo is definitely indexing now when I search it: https://github.com/search?q=geopopos%2Faws_lasso_to_notion

but I'm still getting the same message in serverless cicd.

does the repository name need to match the app name?

should the serverless.yml file be in the root directory or somewhere else? (https://github.com/geopopos/aws_lasso_to_notion)

**Reply by masanaikeshima** [reply]
I gave up with the serverless ci/cd as the delay of GitHub Search API meant it wasn't useful for my situation. 

Try looking to use the GitHub action for deployment, it's worked very well for me and is also free for most use cases with arguably more flexibility.

**Reply by geopopos** [reply]
@masanaikeshima appreciate it! Anywhere you suggest I start?

**Reply by masanaikeshima** [reply]
Use this as a basis, does take a bit of research but it does work:
https://github.com/serverless/github-action

**Comment by masanaikeshima** [comment]
Same boat, just signed up for the teams tier as all our repositories are in an organisation account and are private. I have tried private and public repositories, and using personal / organisation repositories as well as the basic node example and no luck. 

Reached out to support, but they might take up to a day to feedback. Any help would be appreciated!

**Reply by geopopos** [reply]
dang that's disappointing :(

**Reply by masanaikeshima** [reply]
I did hear back from the support:

Hi there. Usually, when a new repo is connected the GitHub Search API indexes the new repo and this has usually taken at most a few minutes. For some reason this TTL to index (or reindex) repos has been increased so it may take some time for the Dashboard CI/CD to query your repo and get results from it. If you give it some more time it should start working as intended.

It can affect all repositories. When you add a new repo or want to add CI/CD for a new service added to an existing repo, GitHub still needs to index that repo in order for our API calls to return the file data we need in order to determine the contents of the repo. There is no way for us to configure a folder in a monorepo as a new service to be deployed in CI/CD if GitHub isn't showing us those folders.
 
This is an issue we have seen come and go for a few months now and looking at alternatives do not really provide any viable options. Over the course of the next few days this "latency" from GitHub should go away like it always has.

I've went back for more details but it's not sounding promising unfortunately.

--- 

I've been using Vercel for the past year or so with a flawless CI/CD process, really hope Serverless works as this is not working.

**Reply by masanaikeshima** [reply]
Just to add, I can see that if you do use the GitHub search and search for your repository "repo:AAA/BBB" then you might also see the same error message:

![CleanShot 2022-07-19 at 09 13 10](https://user-images.githubusercontent.com/7580706/179701443-9a65ffcf-899d-4a99-b599-36efc530aa0b.png)

As Serverless CI/CD is dependant on this API until this message changes it probably won't work and it can take a while for this to get indexed. Surprised that this the case as Vercel doesn't have any issues with newly created repositories and hopefully this can change.

**Reply by geopopos** [reply]
Thanks for the update @masanaikeshima !
